kernel/src/net/socket/mod.rs

```diff
@@ -17,8 +20,42 @@ pub mod unix;
 mod util;
 pub mod vsock;
 
+mod private {
+    use crate::{events::IoEvents, prelude::*, process::signal::Pollable};
+
+    /// Common methods for sockets, but private to the network module.
+    ///
+    /// These are implementation details of sockets, so shouldn't be accessed outside the network
+    /// module. Therefore, the whole trait is sealed.
+    pub trait SocketPriv: Pollable {
+        /// Returns whether the socket is in non-blocking mode.
+        fn is_nonblocking(&self) -> bool;
+
+        /// Sets whether the socket is in non-blocking mode.
+        fn set_nonblocking(&self, nonblocking: bool);
+
+        /// Blocks until some events occur to complete I/O operations.
+        ///
+        /// If the socket is in non-blocking mode and the I/O operations cannot be completed
+        /// immediately, this method will fail with [`Errno::EAGAIN`] instead of blocking.
+        ///
+        /// [`Errno::EAGAIN`]: `crate::Errno::EAGAIN`
```

Comment:
The error may be better refactored as

```rust
/// [`EAGAIN`]: crate::error::Errno::EAGAIN
```

to be consistent with exisiting code.